docs/DeveloperGuide.md

```diff
@@ -1,34 +1,956 @@
+
+Step 3. `AddCommandTimetableMode#execute()` is called, which then calls `InputParser#parseTaskTitle`, which first extracts the `title` from the user's input.
+
+Step 4. `InputParser#parseDeadline` is then called, which returns the `deadline` to `AddCommandTimetableMode#execute()`.
+
+Step 5. `TaskList#addTask` is then called and a new `Task`, with `title` and `deadline`, is initialised.
+
+Step 6. To signal that the user has successfully added a task, a message is printed with `CliMessages#printAddedTaskMessage`.
+
+The UML sequence diagram below shows how the add task command works.
+![Sequence Diagram for Add Task Command](/diagrams/class/jpeg/add_task.jpg)
+
+<hr>
+
+The figure below shows how the delete task command works:
+<img src= "https://user-images.githubusercontent.com/60319628/96657942-02dc6900-1376-11eb-9284-38322e1a2b09.png">
+
+1. The `CliUserInterface` receives the "delete 1" input by the user and passes it to the `InputParser` class.
+2. `InputParser` parses the input to determine the type of command and the index of the task that is required to delete.
+The Parser then constructs a `RemoveCommandTimetableMode` with constructor as shown below.
+```java
+public RemoveCommandTimetableMode(int indexToRemove, AppState uiMode) {
+    this.setAppState(uiMode);
+    this.indexToRemove = indexToRemove;
+}
+```
+3. Method **execute()** then calls the `TaskList` stored in `AppState` to update the deletion of the task.
+It also constructs `CliMessages` to display messages to the user.
+4. If the deletion is successful, `CliMessages` displays the message to the user.
+
+##### 4.2.1.2. Design Considerations
+
+###### Aspect: How to store tasks in `TaskList`
+- **Alternative 1 (current choice):** Store as an `ArrayList` of tasks
+    - Pros: It is easier to implement because the code base are list based.
+    - Cons: It is unoptimized in terms of complexity, which requires more work for scaling of the application.
+- **Alternative 2:** Store as a Hash Table with the key as the index and value as `Task`
+    - Pros: It has a better time complexity and reduce the work in scaling stage since this data structure is more optimized (O(1) can be achieved).
+    - Cons: It takes more resources to implement. The constant factor for a hashing algorithm is significant and not
+     worth the tradeoff for smaller amounts of data, like a typical user would be likely to generate.
+
+#### 4.2.2. Tag Feature
+The user can tag `Task`s in the `TaskList`. This section describes the implementation and design considerations for this
+feature.
+
+##### 4.2.2.1. Implementation
+The `Task` class contains a member `tag` of String type.
+
+The figure below shows how the tag operation works:
+
+![Sequence Diagram for Tag Timetable command](/diagrams/class/jpeg/SequenceDiagram_TagTimetable.jpg)
```

Comment:
Same as the previous diagram, should the lines be a solid line instead?
<img width="322" alt="2020-10-28-131555" src="https://user-images.githubusercontent.com/12934891/97395353-bec20900-191f-11eb-967d-2107d5a2bfce.png">

Reply:
Agreed.